README
NAME
    Acme::Alien::DontPanic - Test Module for Alien::Base

DESCRIPTION
    Alien::Base comprises base classes to help in the construction of
    "Alien::" modules. Modules in the Alien namespace are used to locate and
    install (if necessary) external libraries needed by other Perl modules.

    This module is a toy module to test the efficacy of the Alien::Base
    system. This module is depended on by another toy module
    Acme::Ford::Prefect, which needs the libdontpanic library to be able to
    tell us the "answer".

SEE ALSO
    *   Alien::Base

    *   Alien

    *   Acme::Ford::Prefect

SOURCE REPOSITORY
    <http://github.com/Perl5-Alien/Alien-Base-Extras>

AUTHOR
    Original author: Joel Berger, <joel.a.berger@gmail.com>

    Current maintainer: Graham Ollis <plicease@cpan.org>

COPYRIGHT AND LICENSE
    Copyright (C) 2012 by Joel Berger

    This library is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.



Changes
Revision history for Perl {{$dist->name}}

{{$NEXT}}

2.7200    2022-10-26 06:30:12 -0600
  - Require Alien::Build 2.72

2.6500    2022-08-30 11:21:25 -0600
  - Require Alien::Build 2.65

2.6400    2022-08-30 10:17:13 -0600
  - Require Alien::Build 2.64

2.5900    2022-08-16 09:50:03 -0600
  - Require Alien::Build 2.59
  - Check cryptographic signature via Digest::Negotiate (gh#11)

2.5300    2022-08-12 14:26:36 -0600
  - Require Alien::Build 2.53

2.5100    2022-08-03 09:58:42 -0600
  - Require Alien::Build 2.51

2.2900    2020-08-28 10:31:53 -0600
  - Require Alien::Build 2.29
  - Migrate inline tests from ABMB

2.2600    2020-08-12 04:15:50 -0600
  - Require Alien::Build 2.26
  - Migrated from Alien::Base::ModuleBuild to Alien::Build::MM

2.1100    2020-03-09 03:09:37 -0600
  - Require Alien::Build 2.11 Alien::Base::ModuleBuild 1.14

2.0400    2020-02-05 19:09:04 -0700
  - Require Alien::Build 2.04 Alien::Base::ModuleBuild 1.14

2.0000    2020-02-02 06:27:17 -0700
  - Require Alien::Build 2.00 Alien::Base::ModuleBuild 1.12

1.9800    2020-01-30 07:59:10 -0700
  - Require Alien::Build 1.98 Alien::Base::ModuleBuild 1.10
  - Switched to Dist::Zilla for speedy releases.

1.96 2020-01-28 23:35:59 -0700
  - Require Alien::Base::ModuleBuild 1.08
  - Require Alien::Base 1.96

1.03 2018-05-14 16:31:56 -0400
  - Switch to new alienfile.org for source
  - Require Alien::Base::ModuleBuild 1.03
  - Require Alien::Base 1.41

0.044 2017-08-23 13:46:23 -0400
  - Require Alien::Base 1.01
  - Require Alien::Base::ModuleBuild 0.044

0.042 2017-07-30 06:18:20 -0400
  - Require Alien::Base 0.75
  - Requrre Alien::Base::ModuleBuild 0.042

0.038  Mar 6, 2017
  - Require Alien::Base 0.038

0.032  Feb 9, 2016
  - Require Alien::Base 0.032

0.031  Jan 31, 2016
  - Require Alien::Base 0.031

0.030  Nov 2, 2016
  - Require Alien::Base 0.030
  - No longer bundle dontpanic with this dist, instead download
    it from the official release site:
      https://perl5-alien.github.io/page/dontpanic.html

0.029  Jun 22, 2016
  - Require Alien:Base 0.029

0.028  Mar 14, 2016
  - Require Alien::Base 0.028

0.027  Jan 20, 2016
  - Require Alien::Base 0.025

0.026  Jan 10, 2016
  - Require Alien::Base 0.024
  - Added t/xs.t and t/ffi.t tests that use Test::Alien

0.025  Jul 20, 2015
  - Require Alien::Base 0.022

0.024  Jul 13, 2015
  - Require Alien::Base 0.020

0.023  Apr 27, 2015
  - Bumped version to match new version of Acme::Ford::Prefect

0.022  Apr 24, 2015
  - Require Alien::Base 0.017

0.021  Apr 22, 2015
  - Require Alien::Base 0.016
  - use alien_stage_install

0.020  Mar 17, 2015
  - Require Alien::Base 0.015
  - Fix for static test on MSWin32 and FreeBSD
  - Remove extranious -Wall flag from AM_CFLAGS that was causing trouble
    on Solaris (and probably other Perls not built by gcc)

0.019  Feb 26, 2015
  - Additional diagnostics during testing
  - Add -I${prefix} to dontpanic .pc file for bib install

0.018  Feb 25, 2015
  - Require Alien::Base 0.014

0.017  Feb 24, 2015
  - Release identical to 0.016
  - Intended for use with Acme::Ford::Prefect 0.017

0.016  Feb 22, 2015
  - Require Alien::Base 0.012

0.015  Feb 22, 2015
  - Require Alien::Base 0.011

0.014  Feb 17, 2015
  - Require Alien::Base 0.010

0.012  Feb 6, 2015
  - Require Alien::Base 0.009

0.011  Jan 8, 2015
  - Require Alien::Base 0.007

0.010  Jan 8, 2015
  - add test to make sure that we produce a static lib that can be used to create a dynamic lib

0.009  Oct 14, 2014
  - Require Alien::Base 0.006

0.008  Oct 9, 2014
  - bundle a dontpanic that includes a dll on windows

0.007  Sep 22, 2014
  - use alien_inline_auto_include for Inline integration

0.006  Sep 11, 2014
  - Require Alien::Base 0.005
  - Support for building system provided dontpanic (plicease gh#2)

0.005  Mar 4, 2013
	- Depend on Alien::Base 0.003

0.003  Nov 4, 2012
	- Move more tests from here to Acme::Ford::Prefect
	- Depend on Alien::Base 0.001_001

0.002  Oct 8, 2012
	- Fixes related to Alien::Base 0.000_022 dev release (major refactoring)

0.001  Jul 30, 2012
	- Initial release



Build.PL
use strict;
use warnings;

use Alien::Base::ModuleBuild;

if($^O eq 'midnightbsd')
{
  print "Sorry, MidnightBSD is not supported.\n";
  print "See:\n";
  print "https://github.com/Perl5-Alien/Alien-Base-Extras/issues/4\n";
  print "pull requests to fix this are welcome.\n";
  print "OS Unsupported\n";
  exit;
}

my $builder = Alien::Base::ModuleBuild->new(
  module_name => 'Acme::Alien::DontPanic',
  license => 'perl',
  configure_requires => {
    'Alien::Base::ModuleBuild' => '0.044',
    'Module::Build' => 0.38,
  },
  test_requires => {
    'Test2::V0'           => 0,
    'ExtUtils::CBuilder' => 0,
    'Capture::Tiny'      => 0,
    'Test::Alien'        => '0.05',
    'Test2'              => '1.302015',
    'Test2::Suite'       => '0.000030',
  },
  requires => {
    'perl' => '5.008001',
    'Alien::Base' => '1.01',
  },
  dist_author => [
    'Graham Ollis <plicease@cpan.org>',
    'Joel A. Berger <joel.a.berger@gmail.com>',
  ],
  alien_name => 'dontpanic',
  alien_isolate_dynamic => 1,
  alien_repository => {
    protocol => 'https',
    host     => 'alienfile.org',
    location => '/dontpanic/',
    pattern  => qr/\/([\d\.]+)\.tar\.gz$/,
  },
  alien_inline_auto_include => [ 'libdontpanic.h' ],
  meta_merge => {
    resources  => {
      repository => "http://github.com/Perl5-Alien/Acme-Alien-DontPanic",
      bugtracker => "http://github.com/Perl5-Alien/Acme-Alien-DontPanic/issues",
      x_MailingList => 'https://groups.google.com/forum/#!forum/perl5-alien',
      x_IRC => "irc://irc.perl.org/#native",
    },
  },  
  alien_stage_install => 1,
);

$builder->add_to_cleanup(
  'inc/dontpanic',
  't/static_link/*.o',
  't/static_link/main.so',
  't/static_link/main.dll',
  't/static_link/main.bundle',
);

$builder->create_build_script;

